Packs/ApiModules/Scripts/AnsibleApiModule/AnsibleApiModule.py

```diff
@@ -0,0 +1,296 @@
+                    username = int_params.get('creds', {}).get('credentials').get('user')
+                    password = int_params.get('creds', {}).get('credentials').get('password')
+
+                    new_host['ansible_user'] = username
+                    new_host['ansible_password'] = password
+
+                # username/password individually entered
+                else:
+                    username = int_params.get('creds', {}).get('identifier')
+                    password = int_params.get('creds', {}).get('password')
+
+                    new_host['ansible_user'] = username
+                    new_host['ansible_password'] = password
+
+                # ios specific
+                if host_type == 'ios':
+                    new_host['ansible_connection'] = 'network_cli'
+                    new_host['ansible_network_os'] = 'ios'
+                    new_host['ansible_become'] = 'yes'
+                    new_host['ansible_become_method'] = 'enable'
+                    inventory['all']['hosts'][host] = new_host
+
+                # nxos specific
+                elif host_type == 'nxos':
+                    new_host['ansible_connection'] = 'network_cli'
+                    new_host['ansible_network_os'] = 'nxos'
+                    new_host['ansible_become'] = 'yes'
+                    new_host['ansible_become_method'] = 'enable'
+                    inventory['all']['hosts'][host] = new_host
+
+            # winrm
+            elif host_type == 'winrm':
+                # Only two credential options
+                # Password saved in credential manager selection
+                if int_params.get('creds', {}).get('credentials').get('password'):
+                    username = int_params.get('creds', {}).get('credentials').get('user')
+                    password = int_params.get('creds', {}).get('credentials').get('password')
+
+                    new_host['ansible_user'] = username
+                    new_host['ansible_password'] = password
+
+                # username/password individually entered
+                else:
+                    username = int_params.get('creds', {}).get('identifier')
+                    password = int_params.get('creds', {}).get('password')
+
+                    new_host['ansible_user'] = username
+                    new_host['ansible_password'] = password
+
+                new_host['ansible_connection'] = "winrm"
+                new_host['ansible_winrm_transport'] = "ntlm"
+                new_host['ansible_winrm_server_cert_validation'] = "ignore"
+
+            inventory['all']['hosts'][host] = new_host
+
+    return inventory, sshkey
+
+
+host_type: str  # Global defined within the integration module. Defined here because https://github.com/python/mypy/issues/5732
+
+
+def generic_ansible(integration_name: str, command: str, args: Dict[str, Any], int_params: Dict[str, Any]) -> CommandResults:
```

Comment:
might be right to add a docstring to the methods in this module.

Reply:
Good idea